lms/djangoapps/grades/tasks.py

```diff
@@ -71,36 +70,44 @@ def recalculate_subsection_grade_v2(**kwargs):
             was queued so that we can verify the underlying data update.
         score_deleted (boolean): indicating whether the grade change is
             a result of the problem's score being deleted.
-        event_transaction_id(string): uuid identifying the current
+        event_transaction_id (string): uuid identifying the current
             event transaction.
-        event_transaction_type(string): human-readable type of the
+        event_transaction_type (string): human-readable type of the
             event at the root of the current event transaction.
+        score_db_table (ScoreDatabaseTableEnum): database table that houses
+            the changed score. Used in conjunction with expected_modified_time.
     """
     try:
         course_key = CourseLocator.from_string(kwargs['course_id'])
         if not PersistentGradesEnabledFlag.feature_enabled(course_key):
             return
 
-        score_deleted = kwargs['score_deleted']
         scored_block_usage_key = UsageKey.from_string(kwargs['usage_id']).replace(course_key=course_key)
-        expected_modified_time = from_timestamp(kwargs['expected_modified_time'])
 
         # The request cache is not maintained on celery workers,
         # where this code runs. So we take the values from the
         # main request cache and store them in the local request
         # cache. This correlates model-level grading events with
         # higher-level ones.
-        set_event_transaction_id(kwargs.pop('event_transaction_id', None))
-        set_event_transaction_type(kwargs.pop('event_transaction_type', None))
+        set_event_transaction_id(kwargs.get('event_transaction_id'))
+        set_event_transaction_type(kwargs.get('event_transaction_type'))
 
         # Verify the database has been updated with the scores when the task was
         # created. This race condition occurs if the transaction in the task
         # creator's process hasn't committed before the task initiates in the worker
         # process.
-        if not _has_database_updated_with_new_score(
-                kwargs['user_id'], scored_block_usage_key, expected_modified_time, score_deleted,
-        ):
-            raise _retry_recalculate_subsection_grade(**kwargs)
+        if task_func == recalculate_subsection_grade_v2:
```

Comment:
I prefer to be explicit.  

1. It allows us to monitor when previous versions of the tasks are no longer queued.
2. Keeping code version-dependent instead of passively defensive prevents unexpected bugs from creeping in.
3. The explicit versioning changes can be tracked via git changes - potentially useful for open-edx operators.